### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by dobooth**
## Purpose of this PR
- #4286
- #4287
- #4180
- #4168
- #4213
- #4266
- #4254
- #4291

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=4292) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=4292) before we can accept your contribution.<br/>**7** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: abrarpathan19<br/>:white_check_mark: dmytro-ch<br/>:white_check_mark: jeff-matthews<br/>:white_check_mark: belbiy<br/>:white_check_mark: yogeshsuhagiya<br/>:white_check_mark: dobooth<br/>:white_check_mark: ansari-krish<br/>:x: ImgBotApp

File: guides/v2.1/extension-dev-guide/intro/developers_roadmap.md [hunk 1]
```diff
@@ -10,20 +10,22 @@ redirect_from:
...
+To satisfy the minimum required elements for creating or customizing your Magento application:
 
-*	Minimum required elements:
+* [Declare component dependencies]({{ page.baseurl }}/extension-dev-guide/build/composer-integration.html) in `composer.json`.
 
-	*	[Declare component dependencies]({{ page.baseurl }}/extension-dev-guide/build/composer-integration.html) in `composer.json`.
-	*	[Register]({{ page.baseurl }}/extension-dev-guide/build/component-registration.html) the component using `registration.php`.
-	*	Component-specific XML definition files:
+	{:.bs-callout .bs-callout-tip}
+	While you can manage dependencies on your own, it is a recommended and strongly encouraged best practice to use the `composer.json`.
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/intro/developers_roadmap.md]:
```suggestion
	While you can manage dependencies on your own, it is a recommended and strongly encouraged best practice to use the `composer.json` file.
```

File: guides/v2.1/extension-dev-guide/intro/developers_roadmap.md [hunk 2]
```diff
@@ -10,20 +10,22 @@ redirect_from:
...
 
-		*	Modules: [`module.xml`]({{ page.baseurl }}/extension-dev-guide/build/create_component.html)
-		*	Themes: [`theme.xml`]({{ page.baseurl }}/frontend-dev-guide/themes/theme-create.html#fedg_create_theme_how-to_declare)
-		*	Language packages: [`language.xml`]({{ page.baseurl }}/config-guide/cli/config-cli-subcommands-i18n.html#config-cli-subcommands-xlate-pack-meta-xml)
-*	Distribute your component:
-
-	*	[Package your component]({{ page.baseurl }}/extension-dev-guide/package/package_module.html) in `.zip` format.
+* [Register]({{ page.baseurl }}/extension-dev-guide/build/component-registration.html) the component using `registration.php`.
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/intro/developers_roadmap.md]:
```suggestion
* [Register]({{ page.baseurl }}/extension-dev-guide/build/component-registration.html) the component using the `registration.php` file.
```

File: guides/v2.1/frontend-dev-guide/layouts/layout-overview.md [hunk 3]
```diff
@@ -78,26 +78,28 @@ To make the necessary changes, create [extending]({{ page.baseurl }}/frontend-de
 
 The Magento application processes layout files in the following order:
 
-1.	Collects all layout files from modules. The order is determined by the modules order in the module list from `app/etc/config.php`.
-2.	Determines the sequence of [inherited]({{ page.baseurl }}/frontend-dev-guide/themes/theme-inherit.html) themes `[<parent_theme>, ..., <parent1_theme>] <current_theme>`
-3.	Iterates the sequence of themes from last ancestor to current:
+1. Module base files loaded.
+2. Module area files loaded.
+3. Collects all layout files from modules. The order is determined by the modules order in the module list from `app/etc/config.php`. (If their priorities are equal, they are sorted according to their alphabetical priority.)
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/layouts/layout-overview.md]:
```suggestion
3. Collects all layout files from modules. The order is determined by the modules order in the module list from the `app/etc/config.php` file. (If their priorities are equal, they are sorted according to their alphabetical priority.)
```

File: guides/v2.1/frontend-dev-guide/responsive-web-design/rwd-breakpoints.md [hunk 4]
```diff
@@ -46,7 +48,10 @@ To do this, take the following steps:
 1. Copy the `_responsive.less` file to your `<your_theme_dir>/web/css/source/lib/` directory from one of the following locations:
 	- `<your_parent_theme_dir>/web/css/source/lib/_responsive.less`: overriding `_responsive.less` in the parent theme. If there's no such file or no parent theme, use the other option. 
 	- `<your_theme_dir>/web/css/source/lib/_responsive.less`: the library file.
-2. In your `_responsive.less`, add the `.media-width` {% glossarytooltip 1a305bdb-9be8-44aa-adad-98758821d6a7 %}mixin{% endglossarytooltip %} rule for your breakpoint in the corresponding section (desktop or mobile, depending on the type of breakpoint you add).
+2. In your `_responsive.less`, add the `.media-width` {% glossarytooltip 1a305bdb-9be8-44aa-adad-98758821d6a7 %}mixin{% endglossarytooltip %} rule for your breakpoint in the corresponding section (desktop or mobile, depending on the type of breakpoint you add).
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/responsive-web-design/rwd-breakpoints.md]:
```suggestion
2. In your `_responsive.less` file, add the `.media-width` {% glossarytooltip 1a305bdb-9be8-44aa-adad-98758821d6a7 %}mixin{% endglossarytooltip %} rule for your breakpoint in the corresponding section (desktop or mobile, depending on the type of breakpoint you add). 
```

File: guides/v2.1/frontend-dev-guide/responsive-web-design/rwd-breakpoints.md [hunk 5]
```diff
@@ -46,7 +48,10 @@ To do this, take the following steps:
 1. Copy the `_responsive.less` file to your `<your_theme_dir>/web/css/source/lib/` directory from one of the following locations:
 	- `<your_parent_theme_dir>/web/css/source/lib/_responsive.less`: overriding `_responsive.less` in the parent theme. If there's no such file or no parent theme, use the other option. 
 	- `<your_theme_dir>/web/css/source/lib/_responsive.less`: the library file.
-2. In your `_responsive.less`, add the `.media-width` {% glossarytooltip 1a305bdb-9be8-44aa-adad-98758821d6a7 %}mixin{% endglossarytooltip %} rule for your breakpoint in the corresponding section (desktop or mobile, depending on the type of breakpoint you add).
+2. In your `_responsive.less`, add the `.media-width` {% glossarytooltip 1a305bdb-9be8-44aa-adad-98758821d6a7 %}mixin{% endglossarytooltip %} rule for your breakpoint in the corresponding section (desktop or mobile, depending on the type of breakpoint you add). 
+
+{:.bs-callout .bs-callout-info}
+`@media-target` option may have one of the following values: `all`, `desktop` or `mobile`.
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/responsive-web-design/rwd-breakpoints.md]:
```suggestion
The `@media-target` option may have one of the following values: `all`, `desktop` or `mobile`.  
```

File: guides/v2.2/frontend-dev-guide/css-topics/css-preprocess.md [hunk 6]
```diff
@@ -148,6 +148,8 @@ You can find the detailed information about the configuration and other options
 
 In client-side compilation mode, most of the stylesheet customizations display immediately after you reload a page in a browser.
 
+With client-side compilation mode enabled, LESS files are compiled on every page load. This reduces page loading performance.
```
Comment (jeff-matthews) [inline guides/v2.2/frontend-dev-guide/css-topics/css-preprocess.md]:
```suggestion
With client-side compilation mode enabled, LESS files are compiled on every page load. This reduces page-loading performance.
```

File: guides/v2.2/frontend-dev-guide/layouts/layout-overview.md [hunk 7]
```diff
@@ -77,15 +77,17 @@ To make the necessary changes, create [extending][extend] and [overriding][overr
 
 The Magento application processes layout files in the following order:
 
-1.	Collects all layout files from modules. The order is determined by the modules order in the module list from `app/etc/config.php`.
-2.	Determines the sequence of [inherited] themes `[<parent_theme>, ..., <parent1_theme>] <current_theme>`
-3.	Iterates the sequence of themes from last ancestor to current:
+1. Module base files loaded.
+2. Module area files loaded.
+3. Collects all layout files from modules. The order is determined by the modules order in the module list from `app/etc/config.php`. (If their priorities are equal, they are sorted according to their alphabetical priority.)
```
Comment (jeff-matthews) [inline guides/v2.2/frontend-dev-guide/layouts/layout-overview.md]:
```suggestion
3. Collects all layout files from modules. The order is determined by the modules order in the module list from the `app/etc/config.php` file. (If their priorities are equal, they are sorted according to their alphabetical priority.)
```

**Comment by dobooth** [comment]
running tests

**Pull request merged**